File: src/main/java/br/com/controle/financeiro/model/repository/TransactionRepository.java
package br.com.controle.financeiro.model.repository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

import br.com.controle.financeiro.model.entity.Transaction;
import br.com.controle.financeiro.model.entity.UserEntity;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface TransactionRepository extends JpaRepository<Transaction, UUID> {
    List<Transaction> findAllByOwner(UserEntity owner);

    Optional<Transaction> findByIdAndOwner(UUID responsible, UserEntity owner);

}

File: src/main/kotlin/br/com/controle/financeiro/model/repository/TransactionRepository.kt
package br.com.controle.financeiro.model.repository

import java.util.UUID
import java.util.Optional
import br.com.controle.financeiro.model.entity.UserEntity
import br.com.controle.financeiro.model.entity.Transaction
import org.springframework.data.jpa.repository.JpaRepository

interface TransactionRepository : JpaRepository<Transaction, UUID> {
    fun findAllByOwner(owner: UserEntity): List<Transaction>
    fun findByIdAndOwner(responsible: UUID, owner: UserEntity): Optional<Transaction>
}
